### PR TITLE
SONARJAVA-516 Rule: Boolean checks should not be inverted

### PR DESCRIPTION
**Pull request by mpaladin**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    // The logic could be simplified if kind would be exposed in the ST
+    if (expression.is(
+        Tree.Kind.EQUAL_TO, Tree.Kind.NOT_EQUAL_TO,
+        Tree.Kind.LESS_THAN, Tree.Kind.GREATER_THAN,
+        Tree.Kind.LESS_THAN_OR_EQUAL_TO, Tree.Kind.GREATER_THAN_OR_EQUAL_TO)) {
+      context.addIssue(tree, this, "Use the opposite operator (\"" + OPERATORS.get(((BinaryExpressionTree) expression).operatorToken().text()) + "\") instead.");
+    }
+  }
+
+  private static ExpressionTree removeParenthesis(ExpressionTree tree) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java]: This one should really make its way (in a dedicated commit) into a utility class for now. It has already been implemented more than necessary.

File: java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.LOGICAL_COMPLEMENT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    ExpressionTree expression = removeParenthesis(((UnaryExpressionTree) tree).expression());
+    // The logic could be simplified if kind would be exposed in the ST
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java]: I am not really sure this comment is relevant here.

**Comment by benzonico** [comment]
Except for small comments : LGTM!

File: java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java (hunk 2 above)
Comment (mpaladin) [inline java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java]: Removed.

File: java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java (hunk 1 above)
Comment (mpaladin) [inline java-checks/src/main/java/org/sonar/java/checks/BooleanInversionCheck.java]: Fully agree!

**Pull request closed without merging**